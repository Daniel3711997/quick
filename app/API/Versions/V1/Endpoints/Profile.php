<?php

declare(strict_types=1);

namespace Quick\API\Versions\V1\Endpoints;

use WP_REST_Server;
use WP_REST_Response;
use Quick\API\Versions\V1\API;
use Quick\Interfaces\API as APIInterface;

class Profile extends API implements APIInterface {
    public string $route = 'profile';

    public function options(): array {
        return [
            'callback' => [$this, 'getProfile'],
            'methods' => WP_REST_Server::READABLE,
            'permission_callback' => '__return_true',
        ];
    }

    public function getProfile(): WP_REST_Response {
        return $this->createResponse([
            'id' => 1234,
            'name' => 'John Doe',
        ]);
    }
}
