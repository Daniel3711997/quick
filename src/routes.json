{
    "$schema": "../.quick/schema.json",

    "routes": [
        {
            "app": "vanilla",
            "tests": [
                {
                    "function": "__return_true"
                }
            ],
            "entry": {
                "name": "vanilla",
                "path": "src/routes/Vanilla/index.ts"
            }
        },

        {
            "app": "react",
            "server": {
                "key": "profile",
                "method": "getUserProfileData"
            },
            "tests": [
                {
                    "function": "is_page_template",
                    "arguments": ["templates/profile.php"]
                }
            ],
            "entry": {
                "name": "profile",
                "path": "src/routes/Profile/index.ts"
            },
            "rewrites": [
                {
                    "after": "top",
                    "queryVars": ["profile-id"],
                    "regex": "^profile/([0-9]+)/?$",
                    "query": "index.php?pagename=profile&profile-id=$matches[1]"
                }
            ]
        }
    ]
}
