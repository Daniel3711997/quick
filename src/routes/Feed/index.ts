import { createContainer } from 'framework/quick';

import { Feed } from './Feed';

const renderContainer = createContainer({
    container: Feed,
    element: 'rootProfile',
});

renderContainer();
