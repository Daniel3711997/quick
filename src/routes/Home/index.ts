import { createContainer } from '@app/framework/quick';

import { Home } from './Home';

const renderContainer = createContainer({
    container: Home,
    element: 'rootProfile',
});

renderContainer();
