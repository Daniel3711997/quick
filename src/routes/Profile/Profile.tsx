export const Profile = () => {
    return (
        <div>
            <h1>Hello World (Profile)</h1>
        </div>
    );
};
