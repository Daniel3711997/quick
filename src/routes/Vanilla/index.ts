/* eslint-disable import/no-unresolved */

import { Dispose } from 'helpers/dispose';
import { loadSuspense } from 'helpers/suspense';

import 'styles/main.scss';

if ('development' !== import.meta.env.MODE) {
    // @ts-expect-error - No types
    import('vite/modulepreload-polyfill');
}

loadSuspense();

if (import.meta.hot) {
    // (HMR) Hot Module Reload - https://vitejs.dev/guide/api-hmr.html
    import.meta.hot.accept();

    // Cleanup
    import.meta.hot.dispose(() => {
        Dispose.dispose();
    });
}

console.log('Hello World (Vanilla)');
