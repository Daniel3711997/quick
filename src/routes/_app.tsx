import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
import { type ComponentType, StrictMode } from 'react';

import { RequestBoundary } from 'components/RequestBoundary';

interface IAppProps {
    Component: ComponentType;
}

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            staleTime: 1000 * 60 * 20, // 20 minutes
            cacheTime: 1000 * 60 * 60 * 2, // 2 hours
        },
    },
});

export const App = ({ Component, ...pageProps }: IAppProps) => {
    return (
        <StrictMode>
            <QueryClientProvider client={queryClient}>
                <ReactQueryDevtools initialIsOpen={false} />

                <RequestBoundary
                    fallback={({ resetErrorBoundary }) => (
                        <div>
                            <p>Something went wrong</p>
                            <button onClick={() => resetErrorBoundary()}>Try again</button>
                        </div>
                    )}
                >
                    <Component {...pageProps} />
                </RequestBoundary>
            </QueryClientProvider>
        </StrictMode>
    );
};
