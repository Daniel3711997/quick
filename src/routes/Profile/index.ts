import { createContainer } from '@app/framework/quick';

import { Profile } from './Profile';

const renderContainer = createContainer({
    container: Profile,
    element: 'rootProfile',
});

renderContainer();
