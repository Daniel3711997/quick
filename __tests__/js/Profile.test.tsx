import { render } from '@testing-library/react';
import { it, expect } from 'vitest';

import { App } from 'routes/_app';
import { Profile } from 'routes/Profile/Profile';

it('renders correctly', () => {
    // prettier-ignore
    const {
        container
    } = render(<App Component={Profile}/>);

    expect(container).toMatchSnapshot();
});
