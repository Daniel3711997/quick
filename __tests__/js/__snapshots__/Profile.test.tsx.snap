// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<div>
  <div>
    <h1>
      Hello World (Profile)
    </h1>
  </div>
</div>
`;
